runtime/src/bank.rs

```diff
@@ -6913,4 +6913,42 @@ mod tests {
             bank = Arc::new(new_from_parent(&bank));
         }
     }
+
+    #[test]
+    fn test_same_program_id_reuses_executable_accounts() {
```

Comment:
I'm having a hard time grokking this test. What is it guarding against exactly? Maybe my confusion starts with the test name. And then what is a "nested processor"?  Why is the nested processor adding a lamport? Isn't the attack about changing "account.data" (didn't want to compile BPF perhaps?)?

Reply:
This issue doesn't have anything specific to do with BPF programs, its about letting nested programs (chain of loaders) re-use the same executable accounts between instructions

Reply:
By "reuse the same", do you mean "reference the same"? Seems like the fix is to remove the reuse.

Reply:
Updated test name to describe better what's being enforced